File: 2015/test/day_03_test.exs
defmodule AoC.Day03Test do
  use ExUnit.Case
  doctest AoC.Day03

  alias AoC.Day03

  describe "part 1" do
    test "sample 1" do
      input = ">"
      assert Day03.part_1(input) == 2
    end

    test "sample 2" do
      input = "^>v<"
      assert Day03.part_1(input) == 4
    end

    test "sample 3" do
      input = "^v^v^v^v^v"
      assert Day03.part_1(input) == 2
    end
  end

  describe "part 2" do
    test "sample 1" do
      input = "^v"
      assert Day03.part_2(input) == 3
    end

    test "sample 2" do
      input = "^>v<"
      assert Day03.part_2(input) == 3
    end

    test "sample 3" do
      input = "^v^v^v^v^v"
      assert Day03.part_2(input) == 11
    end
  end
end

File: 2023/src/days/__tests__/day03.test.ts
import DayModule from '@/days/day03';

describe('part 1', () => {
  test('sample', () => {
    const input = `467..114..
...*......
..35..633.
......#...
617*......
.....+.58.
..592.....
......755.
...$.*....
.664.598..`;

    expect(DayModule.part1(input)).toBe(4361);
  });
});

describe('part 2', () => {
  test('sample', () => {
    const input = `467..114..
...*......
..35..633.
......#...
617*......
.....+.58.
..592.....
......755.
...$.*....
.664.598..`;

    expect(DayModule.part2(input)).toBe(467835);
  });
});
